### ISSUE TITLE
use spring resttemplate or webclient to place REST requests

### ISSUE THREAD

**Issue by patrickp-at-work** [razorpay/razorpay-java#122]
spring rest template [allows better abstraction of rest interfaces](https://spring.io/guides/gs/consuming-rest/#initial) and is a de facto industry standard. Alternatively, there is [Spring webclient](https://www.baeldung.com/spring-5-webclient) now. Why do you use the `OkHttpClient` which is a lower level library? If you try to avoid the transitive dependency to the Spring ecosystem: can you provide a separate client library for Spring?

**Comment by shubhamharitash** [comment]
is this issue still open?

**Comment by madhavanramachandran** [comment]
Yes, i feel the same. We don't have enough control on the logging and handling response. It is a bigger problem for us as the end users are not clear on the error messages given in the exceptions.
